modules/effects/arrow-effects-rx2-data/src/main/kotlin/arrow/effects/rx2/SingleK.kt

```diff
@@ -24,6 +25,13 @@ data class SingleK<A>(val single: Single<A>) : SingleKOf<A>, SingleKKindedJ<A> {
   fun <B> map(f: (A) -> B): SingleK<B> =
     single.map(f).k()
 
+  fun <B> mapFilter(f: (A) -> Option<B>): SingleK<B> =
+    single
+      .filter { f(it).isDefined() }
+      .map { f(it).orNull()!! }
+      .toSingle()
+      .k()
```

Comment:
Same as MonoK, I'm surprised this is a valid instance

Reply:
Changed to `flatMap` as suggested